dnf/base.py

```diff
@@ -91,6 +92,9 @@ def __enter__(self):
         return self
 
     def __exit__(self, *exc_args):
+        if (not self.conf.keepcache and
+                not self.ts.isTsFlagSet(rpm.RPMTRANS_FLAG_TEST)):
+            self.clean_used_packages()
```

Comment:
Please move these three lines into ```Base.close``` since we also support people that prefer the old-fashioned way of using stateful objects (```base = Base()```, ```try:```, ..., ```finally:```, ```base.close()```) in which case the ```__exit__``` method is not called.